tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ValueConverterTests.cs

```diff
@@ -0,0 +1,533 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Runtime.CompilerServices;
+using Microsoft.EntityFrameworkCore;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace MongoDB.EntityFrameworkCore.FunctionalTests.Mapping;
+
+[XUnitCollection("MappingTests")]
+public class ValueConverterTests(TemporaryDatabaseFixture tempDatabase)
+    : IClassFixture<TemporaryDatabaseFixture>
+{
+    class IdIsObjectId
+    {
+        public ObjectId _id { get; set; }
+    }
+
+    class IdIsString
+    {
+        public string _id { get; set; }
+    }
+
+    private static readonly Action<ModelBuilder>? ClrObjectIdToMongoString = mb =>
+    {
+        mb.Entity<IdIsObjectId>()
+            .Property(e => e._id).HasConversion(v => v.ToString(), v => ObjectId.Parse(v));
+    };
+
+    [Fact]
+    public void ObjectId_can_deserialize_from_string()
+    {
+        var expectedId = ObjectId.GenerateNewId();
+        var expected = new IdIsString { _id = expectedId.ToString() };
+        tempDatabase.CreateTemporaryCollection<IdIsString>().InsertOne(expected);
+
+        var collection = GetCollection<IdIsObjectId>();
+        var db = SingleEntityDbContext.Create(collection, ClrObjectIdToMongoString);
+
+        var found = db.Entitites.First();
```

Comment:
Entitites => Entities (extra "t" in DbSet<T> name)

Reply:
I'll resolve that in another PR as it affects a whole bunch of tests and predates the PR.